src/ParallelAlgorithms/Amr/Criteria/TruncationError.hpp

```diff
@@ -0,0 +1,113 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Domain/Amr/Flag.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PowerMonitors.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Options/String.hpp"
+#include "ParallelAlgorithms/Amr/Criteria/Criterion.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t>
+class ElementId;
+/// \endcond
+
+namespace amr::Criteria {
+
+/*!
+ * \brief Refine the grid towards the target truncation error
+ *
+ * If any tensor component has a truncation error above the target value, the
+ * element will be p-refined. For details on how the truncation error is
+ * computed see `PowerMonitors::truncation_error`.
+ *
+ * \tparam Dim Spatial dimension of the grid
+ * \tparam TensorTags List of tags of the tensors to be monitored
+ */
+template <size_t Dim, typename TensorTags>
+class TruncationError : public Criterion {
+ public:
+  struct TargetTruncationError {
```

Comment:
Name this `TargetAbsoluteTruncationError`?

Why not include a relative error too?

Reply:
Added relative and absolute error